db/migrate/20180910155506_add_uniqueness_of_order_id_to_spree_shipments.rb

```diff
@@ -1,5 +1,19 @@
+# This migration is an OFN specific migration that enforces an order to have a single shipment at all times
 class AddUniquenessOfOrderIdToSpreeShipments < ActiveRecord::Migration
   def change
+    # We keep the latest shipment in the order
+    Spree::Shipment
+      .select(:order_id)
+      .group(:order_id)
+      .having("count(*) > 1")
+      .each do |shipment|
+        Spree::Shipment
+          .where("order_id = #{shipment.order_id}")
+          .order("updated_at")
+          .first
```

Comment:
So you destroy only one shipment here, right? Could there be more than one?

I have a different idea:
```rb
latest_shipments = Spree::Shipment.
  select("order_id, MAX(updated_at) updated_at").
  group(:order_id).
  having("count(*) > 1")

Spree::Shipment.
  joins("INNER JOIN (#{latest_shipments.to_sql}) latest_shipments ON spree_shipments.order_id=latest_shipments.order_id AND spree_shipments.last_updated!=latest_shipments.updated_at").
  destroy_all
```

Reply:
accepted, thanks a lot!